libraries/SD/src/SD.cpp

```diff
@@ -350,6 +350,17 @@ boolean SDClass::begin(uint8_t csPin, uint32_t speed) {
          root.openRoot(volume);
 }
 
+// Ends SD card connection
+void SDClass::end() {
+    if(card.errorCode() == 0 && root.isOpen()) {
```

Comment:
Can root be open but card have an errorCode? What happens if root.close() is called regardless of the errorCode? i.e.:
```
if(root.isOpen())
  root.close();
```